### PR TITLE
RSpec changes needed for Neo4j Orm Adapter

### PR DESCRIPTION
**Pull request by andreasronge**
Some RSpec changes (bug fixes) in order to make the neo4j orm_adapter happy.
- Using :offset and :limit does not make sense without using :order in `find_all`
- RSpec 3.0 was released. Fixed one spec so it works both for RSpec 2.x and 3.x

### CODE REVIEW

File: spec/orm_adapter/example_app_shared.rb [hunk 1]
```diff
@@ -134,14 +136,14 @@ def reload_model(model)
           user1 = create_model(user_class, :name => "Fred")
           user2 = create_model(user_class, :name => "Fred")
           user3 = create_model(user_class, :name => "Betty")
-          user_adapter.find_all(:name => "Fred").should == [user1, user2]
+          user_adapter.find_all(:name => "Fred").to_a.should =~ [user1, user2]
```
Comment (josevalim) [inline spec/orm_adapter/example_app_shared.rb:139]: Hello @andreasronge ! How is it going?

Question: do you think .to_a makes sense here? Shouldn't neo4j adapter return an array? Or, to change slightly the question, shouldn't neo4j adapter return the same as the others?

Comment (andreasronge) [inline spec/orm_adapter/example_app_shared.rb:139]: Hi there
Well it does not work for ActiveRecord either and I think we should not assume `find_all` to return an array. (Neo4j takes all my spare time :) so still no time for playing around with elixir :(  )

Without #to_a

```
  1) [ActiveRecord orm adapter] it should behave like example app with orm_adapter adapter instance #find_all (conditions) should return only models matching conditions
     Failure/Error: user_adapter.find_all(:name => "Fred").should =~ [user1, user2]
       expected: [#<ArOrmSpec::User id: 15, name: "Fred", rating: nil>, #<ArOrmSpec::User id: 16, name: "Fred", rating: nil>]
            got: [#<ArOrmSpec::User id: 15, name: "Fred", rating: nil>, #<ArOrmSpec::User id: 16, name: "Fred", rating: nil>] (using =~)
       Diff:
       @@ -1,3 +1,2 @@
       -[#<ArOrmSpec::User id: 15, name: "Fred", rating: nil>,
       - #<ArOrmSpec::User id: 16, name: "Fred", rating: nil>]
       +[#<ArOrmSpec::User id: 15, name: "Fred", rating: nil>, #<ArOrmSpec::User id: 16, name: "Fred", rating: nil>]

     Shared Example Group: "example app with orm_adapter" called from ./spec/orm_adapter/adapters/active_record_spec.rb:36
     # ./spec/orm_adapter/example_app_shared.rb:139:in `block (5 levels) in <top (required)>'

```

**Comment by josevalim** [comment]
It makes sense then, thanks!

For some reason this pull request breaks the build (and the last commit passed just fine). Can you please help me find out what went wrong?

**Comment by andreasronge** [comment]
Looks like it is because it now uses RSpec 3.0. Maybe we should lock Gemfile to RSpec 2.x for now.

**Comment by josevalim** [comment]
Ah, there is no Gemfile lock... we should probably add some locks to the repo.

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Comment by ianwhite** [comment]
Thanks Andreas & José!

btw, The reason there was no Gemfile.lock, was so that we catch errors on travis with newer versions of orms.  Is that a good or a bad idea do you think?

**Comment by josevalim** [comment]
Oh, that's a good point. The downside though is pull requests can fail for unrelated reasons. So I am not sure on which side to err.

**Comment by andreasronge** [comment]
I think it is good. Now we know that it does not work with latest released RSpec :)

**Comment by ianwhite** [comment]
Ok, I'll do some work making sure everything works with the latest and greatest.

**Pull request merged**